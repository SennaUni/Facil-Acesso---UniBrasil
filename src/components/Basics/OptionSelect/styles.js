// import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';


export const Container = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  border-bottom-color: #EEE;
  border-bottom-width: 1px;
  padding: 10px;

  background-color: ${props => props.backGround};
`;

export const Picture = styled.Image`
  height: 50px;
  width: 50px;
  border-radius: 25px;
  margin-right: 12px;
`;

export const Label = styled.Text`
  font-size: 16px;
  color: #555;

  font-weight: ${props => props.fontWeight};
`;
