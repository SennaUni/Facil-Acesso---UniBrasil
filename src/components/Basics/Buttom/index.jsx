import React from 'react';
import { ActivityIndicator } from 'react-native';

import { Container, Title } from './styles';

export function Buttom({ title, loading = false, ...rest }) {
  return (
    <>
      { loading ? (
        <Container {...rest} loading={true} disabled={true}>
          <ActivityIndicator
            size='small'
            color='#FFF'
          />
          <Title>Loading</Title>
        </Container>
      ) : (
        <Container {...rest} >
          <Title>{title}</Title>
        </Container>
      )}
    </>
  );
}