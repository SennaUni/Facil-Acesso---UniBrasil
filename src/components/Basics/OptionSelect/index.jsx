import React from 'react';

import { FontAwesome } from '@expo/vector-icons';

import { Container, Picture, Label } from './styles';

export function OptionSelect({ item, selectedValue, onPress }) {

  return (
    <Container 
      backGround={item.id === selectedValue?.id ? '#EEE' : '#FFF'}
      onPress={onPress}
    >
      <Picture 
        source={{
          uri: item.avatarUrl,
        }}
      />
      <Label 
        fontWeight={item.id === selectedValue?.id ? 'bold' : 'normal'} 
      >
        {item.name}
      </Label>
      {/* { item.id === selectedValue?.id && (
        <FontAwesome
        name='check-square'
        size={22}
        color={'green'}
      />
      )} */}
    </Container>
  );
}
