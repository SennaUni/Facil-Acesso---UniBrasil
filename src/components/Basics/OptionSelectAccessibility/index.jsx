import React from 'react';

import { FontAwesome } from '@expo/vector-icons';

import { Container, Picture, Label } from './styles';

export function OptionSelectAccessibility({ item, selectedValue, onPress }) {

  return (
    <Container 
      backGround={item.id === selectedValue?.id ? '#8241B8' : '#FFF'}
      onPress={onPress}
    >
      <Picture 
        source={{
          uri: item.avatarUrl,
        }}
      />
      <Label 
        color={item.id === selectedValue?.id ? '#FFF' : '#555'}
      >
        {item.name}
      </Label>
      {/* { item.id === selectedValue?.id && (
        <FontAwesome
        name='check-square'
        size={22}
        color={'green'}
      />
      )} */}
    </Container>
  );
}
