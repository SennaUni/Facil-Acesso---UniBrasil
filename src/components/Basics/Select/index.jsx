import React, { useState, useRef, useEffect } from 'react';

import { TouchableOpacity, Modal, FlatList, SafeAreaView } from 'react-native';

import { FontAwesome } from '@expo/vector-icons';

import { useField } from '@unform/core';

import { Label, Container, ComboBox, ModalHeader, SelectTitle, ModalTitle, ModalClose } from './styles';

export function Select({ name, options, text, label, OptionComponent }) {
  const selectRef = useRef(null)

  // const { fieldName, registerField } = useField('teste');

  const [openModal, setOpenModal] = useState(false);
  const [selectedValue, setSelectedValue] = useState();

  function renderOption(item) {
    return (
      <OptionComponent 
        item={item}
        selectedValue={selectedValue}
        onPress={() => {
          setSelectedValue(item);
          setOpenModal(false);
        }}
      />
    )
  }

  // useEffect(() => {
  //   registerField({
  //     name: fieldName,
  //     ref: selectRef.current,
  //     path: 'value',
  //     getValue: ref => {
  //       return ref.value || ''
  //     },
  //     setValue: (ref, value) => {
  //       ref.setNativeProps({ text: value });
  //       ref.value = value
  //     },
  //     clearValue: ref => {
  //       ref.value = ''
  //     },
  //   })
  // }, [fieldName, registerField])

  return (
    <>
      <Label>
        {label}
      </Label>
      <Container >
        <ComboBox
          onPress={() => setOpenModal(true)}
        >
          <SelectTitle 
            numberOfLines={1}
          >
            {selectedValue?.name || text}
          </SelectTitle>
          <FontAwesome
            name='angle-down'
            size={26}
            color={'#000'}
          />
        </ComboBox>
        <Modal 
          animationType='slide'
          visible={openModal}
          onRequestClose={() => setOpenModal(false)}
        >
          <SafeAreaView>
            <ModalHeader>
              <TouchableOpacity
                onPress={() => setOpenModal(false)}
              >
                <FontAwesome
                  name='angle-down'
                  size={30}
                  color={'#000'}
                />
              </TouchableOpacity>
              <ModalTitle>
                {text}
              </ModalTitle>
              <TouchableOpacity
                onPress={() => setOpenModal(false)}
              >
                <ModalClose>
                  Cancelar
                </ModalClose>
              </TouchableOpacity>
            </ModalHeader>
            <FlatList 
              data={options ?? []}
              keyExtractor={item => String(item.id)}
              renderItem={({ item }) => renderOption(item) }
            />
          </SafeAreaView>
        </Modal>
      </Container>
    </>
  );
}