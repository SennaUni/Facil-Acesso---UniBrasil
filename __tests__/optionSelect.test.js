import { render, screen, fireEvent } from '@testing-library/react-native';
import { OptionSelect } from '../src/components/Basics/OptionSelect';

test('Testando conteÃºdo interno do OptionSelect', async () => {
  let buttonPress = false;

  const item = {
    id: 1,
    icon: 'smile-o',
    value: 'Super recomendo'
  };

    const selectedValue = {
      id: 1,
      icon: 'smile-o',
      value: 'Super recomendo'
    };

  function onPressTest() {
    buttonPress = true;
  }

  const { getByText, toJSON } = render(
    <OptionSelect 
      item={item}
      selectedValue={selectedValue}
      onPress={onPressTest}
    />
  );

  fireEvent.press(getByText('Super recomendo'));

  expect(buttonPress).toBe(true);
  expect(getByText('Super recomendo')).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});
