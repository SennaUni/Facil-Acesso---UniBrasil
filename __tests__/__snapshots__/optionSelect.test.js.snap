// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando conte√∫do interno do OptionSelect 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#8241B8",
      "borderRadius": 25,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Text />
  <Text
    color="#FFF"
    style={
      [
        {
          "color": "#FFF",
          "fontSize": 16,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Super recomendo
  </Text>
</View>
`;
